boefjes/boefjes/plugins/kat_rpki/main.py

```diff
@@ -13,39 +13,62 @@
 
 from boefjes.job_models import BoefjeMeta
 
+# Paths and URLs for RPKI
 BASE_PATH = Path(getenv("OPENKAT_CACHE_PATH", Path(__file__).parent))
 RPKI_PATH = BASE_PATH / "rpki.json"
 RPKI_META_PATH = BASE_PATH / "rpki-meta.json"
 RPKI_SOURCE_URL = "https://console.rpki-client.org/vrps.json"
+
+# Paths and URLs for BGP
+BGP_PATH = BASE_PATH / "bgp.jsonl"
+BGP_META_PATH = BASE_PATH / "bgp-meta.json"
+BGP_SOURCE_URL = "https://bgp.tools/table.jsonl"
+
+# Cache timeout and default hash function
 RPKI_CACHE_TIMEOUT = 1800  # in seconds
 HASHFUNC = "sha256"
 
 
 def run(boefje_meta: BoefjeMeta) -> list[tuple[set, bytes | str]]:
     input_ = boefje_meta.arguments["input"]
     ip = input_["address"]
-    now = datetime.utcnow()
     hash_algorithm = getenv("HASHFUNC", HASHFUNC)
 
-    if not RPKI_PATH.exists() or cache_out_of_date():
-        rpki_json, rpki_meta = refresh_rpki(hash_algorithm)
+    # RPKI cache check and refresh
+    if not RPKI_PATH.exists() or cache_out_of_date(RPKI_META_PATH):
+        rpki_json, rpki_meta = refresh_cache(RPKI_SOURCE_URL, RPKI_PATH, RPKI_META_PATH, hash_algorithm)
     else:
-        with RPKI_PATH.open() as json_file:
-            rpki_json = json.load(json_file)
-        with RPKI_META_PATH.open() as json_meta_file:
-            rpki_meta = json.load(json_meta_file)
+        rpki_json = load_json(RPKI_PATH)
+        rpki_meta = load_json(RPKI_META_PATH)
+
+    if not BGP_PATH.exists() or cache_out_of_date(BGP_META_PATH):
+        bgp_data, bgp_meta = refresh_cache(
+            BGP_SOURCE_URL, BGP_PATH, BGP_META_PATH, hash_algorithm, file_extension="jsonl"
+        )
+    else:
+        bgp_data = load_jsonl(BGP_PATH)
+
     exists = False
-    notexpired = False
+    valid = False
     roas = []
+    bgp_entries = []
     for roa in rpki_json["roas"]:
         if IPAddress(ip) in IPNetwork(roa["prefix"]):
-            exists = True
             expires = datetime.fromtimestamp(roa["expires"])
-            roas.append({"prefix": roa["prefix"], "expires": expires.strftime("%Y-%m-%dT%H:%M"), "ta": roa["ta"]})
-            if expires > now:
-                notexpired = True
+            asn = roa["asn"]
+            roas.append(
+                {"prefix": roa["prefix"], "expires": expires.strftime("%Y-%m-%dT%H:%M"), "ta": roa["ta"], "asn": asn}
```

Comment:
I'm not sure if the datetime format and timezone is specified, but you might want to include a timezone in this ISO format?